### PR TITLE
Remove the gutenberg-block-directory experimental flag

### PR DESCRIPTION
**Pull request by tellyworth**
This removes the flag and its UI, enabling the block directory by default.

Note that it does not change the API namespace, which is still `__experimental`.

See #23384.

## How has this been tested?
Manually tested, by verifying that the block directory is searched from the editor:
1. Visit Posts / Add New Post
2. Click Add Block
3. In Search For A Block, type`boxer`, `waves`, or `bliss`.
4. Confirm that some blocks from the directory are shown, headed with the text "No blocks found in your library. These blocks can be downloaded and installed:"

## Types of changes
- Removes conditional checks for the `gutenberg-block-directory` experiment.
- Removes the settings field.
- Removes the related setup from phpunit bootstrap.
- In short, enables the Block Directory by default.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by ellatrix** [comment]
Looks like there's some e2e test failures.

**Comment by StevenDufresne** [comment]
I don’t have your fork setup @tellyworth but we’ll probably want to the remove the following lines:

https://github.com/WordPress/gutenberg/blob/bacb520a0203d1c0d4afe4e9c3f2f090dabb9130/packages/e2e-tests/specs/experiments/block-directory-add.test.js#L14

https://github.com/WordPress/gutenberg/blob/bacb520a0203d1c0d4afe4e9c3f2f090dabb9130/packages/e2e-tests/specs/experiments/block-directory-add.test.js#L176

**Review by noisysocks — APPROVED** [review]
Code looks good and the tests pass! :shipit:

**Review by StevenDufresne — APPROVED** [review]
